### PR TITLE
Add a patch endpoint for /update_project

### PR DESCRIPTION
**Pull request by elsiel23**
To clear the priorities field, specify {'priorities]: []}, not None.

### CODE REVIEW

File: src/planscape/plan/views.py [hunk 1]
```diff
@@ -291,12 +289,29 @@ def update_project(request: HttpRequest) -> HttpResponse:
...
+            _set_project_parameters(max_budget, max_treatment_area_ratio,
+                                    max_road_distance, max_slope, project)
+            _set_priorities(priorities, project)
+            project.save()
+            return HttpResponse(str(project.pk))
+        elif request.method == "PATCH":
+            del body['id']
+            body.pop('priorities', None)
+            s = ProjectSerializer(project, data=body, partial=True)
+            s.is_valid()
```
Comment (strangeties) [inline src/planscape/plan/views.py:304]: Should s.is_valid be called for both PUT and PATCH?

Comment (strangeties) [reply src/planscape/plan/views.py:304]: ah nvm - I see is_valid is for checking the serialization, not for checking project parameters.

File: src/planscape/plan/views.py [hunk 2]
```diff
@@ -291,12 +289,29 @@ def update_project(request: HttpRequest) -> HttpResponse:
...
+            s.save()
+
+            if priorities is not None:
+                project.priorities.clear()
+                _set_priorities(priorities, project)
+            project.save()
```
Comment (strangeties) [inline src/planscape/plan/views.py:310]: nit: to reduce repetition: can project.save() and HttpResponse() be called outside of the if/else control flow?

File: src/planscape/plan/views.py (hunk 1 above)
Comment (strangeties) [inline src/planscape/plan/views.py:304]: I think this should be "s.is_valid(raise_exception=True)"

File: src/planscape/plan/tests.py [hunk 3]
```diff
@@ -478,7 +478,7 @@ def setUp(self):
         self.base_condition = BaseCondition.objects.create(
             condition_name="condition1", condition_level=ConditionLevel.ELEMENT)
         self.condition1 = Condition.objects.create(
-            condition_dataset=self.base_condition)
+            condition_dataset=self.base_condition, condition_score_type=0, is_raw=False)
```
Comment (strangeties) [inline src/planscape/plan/tests.py:481]: nit: is there an enum value that may be used here instead of 0?

File: src/planscape/plan/views.py (hunk 1 above)
Comment (strangeties) [reply src/planscape/plan/views.py:304]: (and please add a test for an invalid body dict)

File: src/planscape/plan/tests.py (hunk 3 above)
Comment (elsiel23) [reply src/planscape/plan/tests.py:481]: Done

File: src/planscape/plan/views.py (hunk 2 above)
Comment (elsiel23) [reply src/planscape/plan/views.py:310]: Done

File: src/planscape/plan/views.py (hunk 1 above)
Comment (elsiel23) [reply src/planscape/plan/views.py]: Done

File: src/planscape/plan/views.py [hunk 4]
```diff
@@ -291,10 +291,27 @@ def update_project(request: HttpRequest) -> HttpResponse:
...
-        project.priorities.clear()
-        _set_project_parameters(max_budget, max_treatment_area_ratio,
-                                max_road_distance, max_slope, priorities, project)
+        
+        if request.method == "PUT":
+            project.priorities.clear()
+            _set_project_parameters(max_budget, max_treatment_area_ratio,
+                                    max_road_distance, max_slope, project)
+            _set_priorities(priorities, project)
+            project.save()
```
Comment (strangeties) [inline src/planscape/plan/views.py:300]: nit: lines 300 and 301 (project.save and return HttpResponse) can be removed

File: src/planscape/plan/tests.py [hunk 5]
```diff
@@ -548,6 +549,86 @@ def test_add_priority(self):
...
+        self.assertEqual(project.max_budget, 100)
+        self.assertEqual(project.max_slope, 0.5)
+        self.assertEqual(project.priorities.count(), 1)
+
+    def test_patch_add_priority(self):
+        self.client.force_login(self.user)
+        project = Project.objects.get(id=self.project_with_user.pk)
+        self.assertEqual(project.max_budget, 100.0)
+        self.assertEqual(project.priorities.count(), 1)
+        response = self.client.patch(
```
Comment (strangeties) [inline src/planscape/plan/tests.py:609]: nit: line 611 could probably be split into two lines

**Review by strangeties — APPROVED** [review]

File: src/planscape/plan/views.py (hunk 4 above)
Comment (elsiel23) [reply src/planscape/plan/views.py:300]: thanks for catching!

**Pull request merged**